src/nisarqa/products/product_reader.py

```diff
@@ -2573,6 +2573,42 @@ def save_browse(
             red=red, green=green, blue=blue, filepath=filepath
         )
 
+    def metadata_geometry_datasets(
+        self,
+    ) -> Iterator[nisarqa.MetadataDataset2D]:
+        """
+        Generator for all metadata datasets in geometry calibration info Group.
+
+        Yields
+        ------
+        ds : nisarqa.MetadataDataset2D
+            The next MetadataDataset2D in this Group:
+                `../metadata/calibrationInformation/geometry`
+        """
+        with h5py.File(self.filepath, "r") as f:
+            grp_path = "/".join([self._calibration_metadata_path, "geometry"])
+            grp = f[grp_path]
+            for ds_arr in grp.values():
+                if isinstance(ds_arr, h5py.Group):
+                    raise TypeError(
+                        f"unexpected HDF5 Group found in {grp_path}."
+                        " Metadata Groups should only contain Datasets."
+                    )
+                ds_path = ds_arr.name
+
+                n_dim = np.ndim(ds_arr)
+                if n_dim in (0, 1):
```

Comment:
Based on the error message below, it looks like we don't expect to see any scalar (i.e. `n_dim == 1`) datasets in this group. If that's the case, then let's change this conditional to only skip 1-D datasets. If we find a 0-D dataset, then the `ValueError` below should be triggered.

Otherwise, we should update the error message below to clarify that 0-D datasets are OK in this metadata group.

```suggestion
                if n_dim == 1:
```

Reply:
I looked into this a bit more, and the metadata groups often contain a `projection` Dataset which is 0-D (aka a scalar), so we do need to keep it as `if n_dim in (0, 1):`.

Accordingly, I updated the error messages to clarify that scalars are ok.

Thanks!